idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixCloudShadowOp.java

```diff
@@ -174,6 +221,34 @@ private int determineSourceResolution(Product product) throws OperatorException
         throw new OperatorException("Invalid product");
     }
 
+    public static double determineResolution(Product product) {
+        int width = product.getSceneRasterWidth();
+        int height = product.getSceneRasterHeight();
+        GeoPos geoPos1 = product.getSceneGeoCoding().getGeoPos(new PixelPos(width / 2, 0), null);
+        GeoPos geoPos2 = product.getSceneGeoCoding().getGeoPos(new PixelPos(width / 2, height - 1), null);
+        double deltaLatInMeters = (geoPos1.lat - geoPos2.lat) / (height-1) / 180.0 * 6367500 * Math.PI;
```

Comment:
You could use  DefaultEllipsoid.WGS84.getSemiMajorAxis() here.

Reply:
See above.